### PR TITLE
Finish Issue 2.3 - Task 2.3

### PR DESCRIPTION
**Pull request by BrianArz**
Task completed as requested.

### CODE REVIEW

File: Task2_3/Lecture.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿namespace Task2_3
+{
```
Comment (IgorKheidorov) [inline Task2_3/Lecture.cs:2]: Think about create base class for all entities to store Description

File: Task2_3/Training.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
+﻿using System.IO;
+using System.Security.Cryptography;
+
+namespace Task2_3
+{
+    internal class Training
+    {
+        public string? Description { get; set; }
```
Comment (IgorKheidorov) [inline Task2_3/Training.cs:8]: Create base class Lectures and Practical Lessons, something like Lesson, and then:
 public Lesson[]? LecturesAndLessons { get; set; }

File: Task2_3/Training.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            }
+
+            return true;
+        }
+
+        public Training Clone()
+        {
+            var clonedTraining = new Training();
+            foreach (var element in LecturesAndLessons)
+            {
```
Comment (IgorKheidorov) [inline Task2_3/Training.cs:60]: Cool, when base class Lesson will be added if ...else is not needed

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]
Cool, fix code according to comments

File: Task2_3/Lecture.cs (hunk 1 above)
Comment (BrianArz) [reply Task2_3/Lecture.cs:2]: Request completed. Waiting for review

File: Task2_3/Training.cs (hunk 2 above)
Comment (BrianArz) [reply Task2_3/Training.cs]: Request Completed. Waiting for review

File: Task2_3/Training.cs (hunk 3 above)
Comment (BrianArz) [reply Task2_3/Training.cs]: Request Completed. Waiting for review

File: Task2_3/Training.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                return false;
+
+            foreach (var element in LecturesAndPractical)
+            {
+                if(element is Lecture) return false;
+            }
+
+            return true;
+        }
+
```
Comment (IgorKheidorov) [inline Task2_3/Training.cs:47]: Sorry, maybe I have forgotten  to comment this previous time, but this is not deep cloning, just just copy references from Trainig to new Training for Lessons, not creating new objects. Please read about deep clone

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]
Sorry, maybe I have forgotten  to comment this previous time, but this is not deep cloning, just just copy references from Trainig to new Training for Lessons, not creating new objects. Please read about deep clone

File: Task2_3/Training.cs (hunk 4 above)
Comment (BrianArz) [reply Task2_3/Training.cs:47]: Requested changes implemented. 

Sorry for the late delivery, I've been quite busy with school.

**Review by IgorKheidorov — APPROVED** [review]

**Pull request merged**